modules/smart_contract_connector/lib/impl.js

```diff
@@ -16,138 +16,246 @@ const OrgId = Contracts.getFromNodeModules('@windingtree/org.id', 'OrgId');
+            provider.on('connect', onConnected);
+        }
+    };
 
+    // Initial connection
+    connect();
+    connectionGuard();
 
     // Cached version of the contracts for lazy loading
     var orgidContract;
     var lifDepositContract;
     var lifTokenContract;
 
     // Get the Organizations in DNS for the DNS Trust Clue
-    const getOrgidFromDns = async (link) => {
-        return new Promise((resolve) => {
-            try {
-                if(link.indexOf('://') === -1) link = `https://${link}`;
-                const myURL = new URL(link);
-                dns.resolveTxt(myURL.hostname, (err, data) => {
-                    if (err) return resolve(undefined);
-                    let orgid = _.get(_.filter(data, (record) => record && record.length && record[0].indexOf('orgid=') === 0), '[0][0]', false);
-                    if (orgid) orgid = orgid.replace('orgid=', '').replace('did:orgid:');
-                    return resolve(orgid);
-                })
-            } catch (e) {
-                resolve(false)
+    const getOrgidFromDns = async (link) => new Promise((resolve) => {
+        try {
+
+            if (link.indexOf('://') === -1) {
+                link = `https://${link}`;
             }
 
-        })
-    };
+            const myURL = new URL(link);
+            dns.resolveTxt(myURL.hostname, (err, data) => {
+
+                if (err) {
+                    return resolve(undefined);
+                }
+
+                let orgid = _.get(
+                    _.filter(
+                        data,
+                        record => record && record.length && record[0].indexOf('orgid=') === 0
+                    ),
+                    '[0][0]',
+                    false
+                );
+
+                if (orgid) {
+                    orgid = orgid.replace('orgid=', '').replace('did:orgid:');
+                }
+
+                return resolve(orgid);
+            });
+        } catch (e) {
+            resolve(false)
+        }
+    });
 
     // Get the Organizations in URL for the Domain Trust Clue
-    const getOrgidFromUrl = async (link) => {
-        return new Promise(async (resolve) => {
-            try {
-                const fetched = await fetch(`${link}/org.id`);
-                let body = await fetched.text();
-                body = body.replace('orgid=', '').replace('did:orgid:');
-                resolve(body);
-            } catch (e) {
-                resolve(false);
-            }
-        })
-    };
+    const getOrgidFromUrl = async (link) => new Promise(async (resolve) => {
+        try {
+            const fetched = await fetch(`${link}/org.id`);
+            let body = await fetched.text();
+            body = body.replace('orgid=', '').replace('did:orgid:');
+            resolve(body);
+        } catch (e) {
+            resolve(false);
+        }
+    });
 
     const getOrgidFromLink = async (link) => {
         let orgid = await getOrgidFromDns(link);
```

Comment:
Not sure of this, "DNS" and "Link" would be different trust assertions no ?

Reply:
I have not touched trust assertions just maybe corrected some styling. I am going to work on trust assertions today.